pkgs/servers/home-assistant/custom-components/xiaomi_miot/package.nix

```diff
@@ -2,29 +2,33 @@
   lib,
   buildHomeAssistantComponent,
   fetchFromGitHub,
+  construct,
   hap-python,
   micloud,
-  pyqrcode,
   python-miio,
   nix-update-script,
 }:
 
 buildHomeAssistantComponent rec {
   owner = "al-one";
   domain = "xiaomi_miot";
-  version = "0.7.21";
+  version = "0.7.23";
 
   src = fetchFromGitHub {
     owner = "al-one";
     repo = "hass-xiaomi-miot";
     rev = "v${version}";
-    hash = "sha256-5MYA5MejQAANyjVqhqZtaIEQEs1K/aOx+1n+L9TmNmY=";
+    hash = "sha256-PTjkKuK+DAOmKREr0AHjFXzy4ktguD4ZOHcWuLedLH0=";
   };
 
-  propagatedBuildInputs = [
+  # Need `hap-python` or:
+  #   File "/nix/store/...-homeassistant-2024.11.1/lib/python3.12/site-packages/homeassistant/components/homekit/__init__.py", line 16, in <module>
+  #    from pyhap import util as pyhap_util
+  # ModuleNotFoundError: No module named 'pyhap'
```

Comment:
That is in a totally different component, the homekit bridge. How does this make sense?

Reply:
Means you need to configure `services.home-assistant.extraComponents = [ "homekit" ]`.

Reply:
With a dependency on homekit home-assistant would install the component dependencies, hadn't we disabled pip.

Reply:
Feel free.